### DISCUSSION TITLE
can cake StartProcess run with "sudo sh"

### DISCUSSION THREAD

**Discussion by hotants** [cake-build/cake#4160] [General]
Hi Team, 
           I have a my.sh to be executed, it's working fine if the file does not require higher permission. 
             StartProcess("sh", new ProcessSettings {
                Arguments = new ProcessArgumentBuilder()
                    .Append($"{myFolder}/my.sh"),
                    WorkingDirectory = myFolder,
                    RedirectStandardOutput = true
            },
                out redirectedStandardOutputUnix
            );

           but it's reporting "permission denied" if try to execute an sh file that requires higher permission. 

           So I tried to use "sodu sh higherpermission.sh" with below code

             StartProcess("sudo sh", new ProcessSettings {
                Arguments = new ProcessArgumentBuilder()
                    .Append($"{myFolder}/higherpermission.sh"),
                    WorkingDirectory = myFolder,
                    RedirectStandardOutput = true
            },
                out redirectedStandardOutputUnix
            );
      
        now it's reporting "An error occurred trying to start process 'sudo sh' with working directory 'myfolder'. No such file or directory"
        could you pls help if StartProcess with sudo, and how to call sudo if it's supported.

Thanks
denny

**Comment by pascalberger** [comment]
Have you tried to just call `sudo` and pass the `sh` in the `ProcessArgumentBuilder` like this:

```
StartProcess(
    "sudo",
    new ProcessSettings {
        Arguments = 
            new ProcessArgumentBuilder()
                .Append("sh")
                .Append($"{myFolder}/higherpermission.sh"),
        WorkingDirectory = myFolder,
        RedirectStandardOutput = true
    },
    out redirectedStandardOutputUnix
);
```

**Reply by hotants** [reply]
Thanks @pascalberger  still reporting the same issue. and does not work if try without WorkingDirectory, or with WorkingDirectory and Append("./higherpermission.sh".
